File: backend/src/api/expenses/Expenses_Model.ts
import { Entity, Column, PrimaryColumn } from 'typeorm';
import { generateUUID } from '../../utils/uuid';

@Entity('expenses')
export class Expenses {
    @PrimaryColumn('uuid')
    id: string;

    @Column()
    name: string;

    @Column('text')
    description: string;

    @Column('double')
    price: number;

    @Column({ name: 'fk_users' })
    fkUser: string;

    @Column()
    payday: Date;

    @Column({ name: 'status_payment', type: 'tinyint' })
    statusPayment: boolean;

    @Column({ name: 'date_created', type: 'timestamp' })
    dateCreated: Date;

    constructor() {
        if (!this.id) {
            this.id = generateUUID();
        }
    }
}

File: app/lib/src/data/expenses/expenses_model.dart
class Expenses {
  final String? id;
  final String name;
  final String description;
  final double price;
  final String fkUser;
  final DateTime payday;
  final bool statusPayment;
  final DateTime? dateCreated;

  Expenses({
    this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.fkUser,
    required this.payday,
    required this.statusPayment,
    this.dateCreated,
  });

  factory Expenses.fromMap(Map<String, dynamic> map) {
    return Expenses(
      id: map['id'],
      name: map['name'],
      description: map['description'],
      price: map['price'],
      fkUser: map['fkUser'],
      payday: map['payday'],
      statusPayment: map['statusPayment'],
      dateCreated: map['dateCreated'],
    );
  }
}
